src/common/webview_chromium.cpp

```diff
@@ -0,0 +1,809 @@
+/////////////////////////////////////////////////////////////////////////////
+// Author: Steven Lamerton
+// Copyright: (c) 2013 Steven Lamerton
+// Licence: wxWindows licence
+/////////////////////////////////////////////////////////////////////////////
+
+#include "wx/wxprec.h"
+
+#if wxUSE_WEBVIEW && wxUSE_WEBVIEW_CHROMIUM
+
+#include "wx/webview.h"
+#include "wx/webview_chromium.h"
+#include "wx/filesys.h"
+#include "wx/rtti.h"
+
+#ifdef __WXMSW__
+#include "wx/msw/private.h"
+#endif
+
+#ifdef __VISUALC__
+#pragma warning(push)
+#pragma warning(disable:4100)
+#endif
+
+#include "include/cef_app.h"
+#include "include/cef_browser.h"
+#include "include/cef_string_visitor.h"
+
+#ifdef __VISUALC__
+#pragma warning(pop)
+#endif
+
+extern WXDLLIMPEXP_DATA_WEBVIEW_CHROMIUM(const char) wxWebViewBackendChromium[] = "wxWebViewChromium";
+
+wxIMPLEMENT_DYNAMIC_CLASS(wxWebViewChromium, wxWebView);
+
+class wxStringVisitor : public CefStringVisitor
+{
+public:
+    enum StringType {
+      PAGE_SOURCE,
+      PAGE_TEXT,
+    };
+    wxStringVisitor(wxWebViewChromium* webview, StringType type) : m_type(type), m_webview(webview) {}
+    void Visit(const CefString& string)
+    {
+        switch(m_type)
+        {
+            case PAGE_SOURCE:
+                 m_webview->SetPageSource(string.ToWString());
+                 break;
+            case PAGE_TEXT:
+                 m_webview->SetPageText(string.ToWString());
+                 break;
+        }
+    }
+private:
+    StringType m_type;
+    wxWebViewChromium *m_webview;
+    IMPLEMENT_REFCOUNTING(wxStringVisitor);
+};
+
+bool wxWebViewChromium::Create(wxWindow* parent,
+           wxWindowID id,
+           const wxString& url,
+           const wxPoint& pos,
+           const wxSize& size,
+           long style,
+           const wxString& name)
+{
+#ifdef __WXMSW__
```

Comment:
What's the difference between the MSW and OSX case? I.e. why don't we check for the return value of `Create()` in the latter and why don't we just have `#ifdef __WXGTK__` and `#else` branches instead of checking for each platform?